vllm/model_executor/layers/fused_moe/layer.py

```diff
@@ -262,38 +281,45 @@ def init_prepare_finalize(self, moe: MoEConfig,
                 # dp_size actually means tp_size, bug in pplx kernels
                 dp_size=all2all_manager.tp_group.world_size,
                 hidden_dim=moe.hidden_dim,
-                hidden_dim_bytes=moe.hidden_dim * moe.in_dtype.itemsize,
+                hidden_dim_bytes=moe.hidden_dim * moe.quant_dtype.itemsize,
                 # For blocked per token: set to
                 #   ceil_div(hidden_dim, block_size) * sizeof(float32)
                 # For per-token: set to sizeof(float32)
-                hidden_dim_scale_bytes=(0 if moe.in_dtype.itemsize != 1 else (
-                    (moe.hidden_dim + moe.block_size - 1) // moe.block_size *
-                    torch.float32.itemsize)),
+                hidden_dim_scale_bytes=(
+                    0 if moe.quant_dtype.itemsize != 1 else
+                    ((moe.hidden_dim + moe.block_size - 1) // moe.block_size *
+                     torch.float32.itemsize)),
                 group_name=all2all_manager.cpu_group.group_name,
             )
 
             handle = all2all_manager.get_handle(all_to_all_args)
 
+            input_activations = get_quant_config_input_activations(
+                quant_config)
```

Comment:
Is this guaranteed to always exist?  I've tried running `Qwen/Qwen3-30B-A3B-FP8` and it didn't appear to have an `input_activations` field in its `quant_config`.

I think this is probably fine for now but maybe with a `None` check.  There's a more general problem of finding the proper quantization info for each MoE layer that needs to be solved.